release.py

```diff
@@ -177,18 +183,66 @@ def push() -> None:
             raise typer.Exit(result)
 
 
-def check_preconditions() -> None:
+def ensure_all_committed_files() -> Optional[List[str]]:
```

Comment:
We are unnecessary strict in this function. We don't need all the files to be committed for the release to happen.
Either everything is committed, and we proceed.
Or, there are files that are not committed, we list them, ask the user to commit them, then ask for confirmation again, but we don't stop the deployment.